lib/hyrax/acl/access_control_list.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Hyrax
+  module Acl
+    READ     = :read
+
+    ##
+    # @api public
+    #
+    # ACLs for `Hyrax::Resource` models
+    #
+    # Allows managing `Hyrax::Permission` entries referring to a specific
+    # `Hyrax::Resource.
+    #
+    class AccessControlList
+      ##
+      # @!attribute [r] access_control
+      #   @return [Hyax:Acl:AccessControl]
+      attr_reader :access_control
+
+      ##
+      # @api public
+      #
+      # @param :access_control [Hyax:Acl:AccessControl]
+      def initialize(access_control:)
+        @access_control = access_control
+      end
+
+      ##
+      # @agent_name [string] greoup_prefx with agent name
+      def has_read_group?(agent_name:)
```

Comment:
Will do. `has_read?` sounds good.

Reply:
what do you think about `agent(agent).has?(:read)`?

Reply:
@no-reply Sorry somehow I missed your comments yesterday. What the  structure of the Hyrax::Agent object looks like, a basic agent object with the agent name only, or something like Hyrax::User? Do we want to handle the group case like Hyrax::Group or not?
If we don't need to handle it more specifically with the Group case, it looks like a general method like `has_grant?(mode:, agent:)` is sufficient to handle any mode extensions.
@marrus-sh What's your thought on that Agent object passing in?

Reply:
@lsitu I think the assumption should be that the agent object implements `#agent_key` and we shouldn’t provide any special handling beyond that. Both `Hyrax::User` ([here](https://github.com/samvera/hyrax/blob/287e9fe95ba897b8b03f37cd10c91e8b59530c06/app/models/concerns/hyrax/user.rb#L56-L61)) and `Hyrax::Group` ([here](https://github.com/samvera/hyrax/blob/287e9fe95ba897b8b03f37cd10c91e8b59530c06/app/models/hyrax/group.rb#L28-L33)) have an `#agent_key` method so they should both work in this case.

Reply:
@marrus-sh Vivian and I just added the Hyrax::Agent class with `#agent_key`. @no-reply We are open to that `agent(agent).has?(:read)` pattern. Just waiting for your comment. Thanks.

Reply:
It might be better to name it `Hyrax::Acl::Agent` so that it doesn’t cause conflicts with upstream Hyrax

We may want a more specific Group implementation for superskunk eventually but I’m not sure that it needs to be in this gem

Reply:
@marrus-sh Thanks for catching that. Vivian and I also port over the Hyrax::Group so that you can use it when needed.